Twitch and youtube integrations

Pena86 opened the pull request.

Pena86 committed:
bot keys default

Pena86 committed:
admin commands to reload modules

samip5 committed:
Added an command error handler, so the error would be sent to user.

Rather than silently erroring.

samip5 committed:
Test command added for Staff.

samip5 committed:
The reload command uses now the new permission system.

samip5 committed:
Command permissions system implemented, the CORRECT way by Discord.py documentation.

https://discordpy.readthedocs.io/en/latest/ext/commands/api.html?highlight=context#checks

Pena86 committed:
Merge pull request #1 from samip5/twitch

Implemented the new permission system

Pena86 committed:
requirements file

samip5 committed:
Squashed commit of the following:

commit c2d270fef8909987128d81b677de6572bb446519
Author: Eero Penttilä <eero.penttila@student.oulu.fi>
Date:   Mon Aug 12 17:57:01 2019 +0300

    requirements file

commit adea5c7df69381c7c39232d41de4f89a4a1f59ac
Merge: cbe411e 22adf3c
Author: Eero Penttilä <Pena86@users.noreply.github.com>
Date:   Mon Aug 12 17:42:50 2019 +0300

    Merge pull request #1 from samip5/twitch

    Implemented the new permission system

samip5 committed:
Added loggers to commands, uses the main bot's logger.

Pena86 committed:
Merge pull request #2 from samip5/twitch

Loggers to commands and changed development prefix

Pena86 committed:
initial twitch module

Pena86 committed:
twitch module almost working

Pena86 committed:
fixed channel polling

Pena86 committed:
post to channel

Pena86 committed:
commands now use channel name or url

Pena86 committed:
alustava youtube

Pena86 committed:
search by name

Pena86 committed:
channels add and remove for following

Pena86 committed:
list videos

samip5 committed:
Added a cat command which will generated a random cat picture to the channel.

samip5 committed:
PEP 8 compliance, https://pep8.org/

samip5 committed:
Merge branch 'random-pictures' into twitch

# Conflicts:
#	bot.py

samip5 committed:
Setting up support for Docker.

More information: https://www.docker.com/get-started

samip5 committed:
Adding uptime command, information about the bot.

Pena86 committed:
youtube latest videos

Pena86 committed:
clean up code

Pena86 committed:
Update youtube.py

samip5 commented:
> Because I cant get it working from inside. Maybe you can?

Can you tell my exactly what do you mean by "I cant get it working"? Does it give you an error code or what is the issue?

The most non-helpful/bad way to explain that something does not work, is to say it just does not work.

Pena86 committed:
Viikkoraportin luominen

alustava ja hyvin puutteellinen versio, joka on kiireessä kasattu ja jollain paikattu jotenkn toimivaksi. Älkää arvostelko :)

Pena86 committed:
Viilausta ja parannusta

Pena86 committed:
many updates to reports

Pena86 committed:
shortcut to reloading extensions

Pena86 committed:
uusia ominaisuuksia ja päivityksiä

Pena86 committed:
new feature picture generation

Pena86 committed:
Merge pull request #4 from Pena86/report

Report
